src/components/protocol_handler/test/protocol_handler_tm_test.cc

```diff
@@ -179,22 +183,48 @@ class ProtocolHandlerImplTest : public ::testing::Test {
     // use protection OFF
     const bool callback_protection_flag = PROTECTION_OFF;
 #endif  // ENABLE_SECURITY
+
     // Expect ConnectionHandler check
-    EXPECT_CALL(session_observer_mock,
-                OnSessionStartedCallback(connection_id,
-                                         NEW_SESSION_ID,
-                                         start_service,
-                                         callback_protection_flag,
-                                         _))
-        .
-        // Return sessions start success
-        WillOnce(Return(session_id));
+    if (waiter) {
+      EXPECT_CALL(session_observer_mock,
+                  OnSessionStartedCallback(connection_id,
+                                           NEW_SESSION_ID,
+                                           start_service,
+                                           callback_protection_flag,
+                                           _))
+          .
+          // Return sessions start success
+          WillOnce(DoAll(NotifyTestAsyncWaiter(waiter), Return(session_id)));
+      if (times) {
+        (*times)++;
+      }
+    } else {
+      EXPECT_CALL(session_observer_mock,
+                  OnSessionStartedCallback(connection_id,
+                                           NEW_SESSION_ID,
+                                           start_service,
+                                           callback_protection_flag,
+                                           _))
+          .
+          // Return sessions start success
+          WillOnce(Return(session_id));
+    }
 
     // Expect send Ack with PROTECTION_OFF (on no Security Manager)
-    EXPECT_CALL(transport_manager_mock,
-                SendMessageToDevice(ControlMessage(FRAME_DATA_START_SERVICE_ACK,
-                                                   PROTECTION_OFF)))
-        .WillOnce(Return(E_SUCCESS));
+    if (waiter) {
+      EXPECT_CALL(transport_manager_mock,
+                  SendMessageToDevice(ControlMessage(
+                      FRAME_DATA_START_SERVICE_ACK, PROTECTION_OFF)))
+          .WillOnce(DoAll(NotifyTestAsyncWaiter(waiter), Return(E_SUCCESS)));
+      if (times) {
+        (*times)++;
+      }
+    } else {
```

Comment:
@istoilovgithub, please rework this method of fixture class, to at least use references, instead of raw pointers. Or rather use "ASSERT_TRUE" to check if data is valid.